### ISSUE TITLE
Missing VAT ExemptionReasonCode

### ISSUE THREAD

**Issue by cgimenez** [ZUGFeRD/mustangproject#345]
Hi all, 
the legitimate (AFAIK) code VATEX-EU-79-C is missing from the schematron rules.

- A relatively recent list is available here : https://ec.europa.eu/digital-building-blocks/wikis/download/attachments/467108974/EN16931%20code%20lists%20values%20v9%20-%20used%20from%202022-03-04.xlsx?version=1&modificationDate=1646394359265&api=v2
- Under the "VATEX" tab, the first entry is
- VATEX-EU-79-C / Exempt based on article 79, point c of Council Directive 2006/112/EC / Repayment of expenditure is not an exemption in the sense of the VAT Directive but may be handled as such in the context of the EN16931.

This code is included in the legitimate codes of French Schematron config files, but not in Mustang ones
As one can see, for the BASIC profile :
https://github.com/ZUGFeRD/mustangproject/blob/866ef8da823390f463cc0139d6bd56faf7d37632/validator/src/main/resources/schematron/ZF_221/FACTUR-X_BASIC_codedb.xml#L4496

The list starts with VATEX-EU-132 which is the second row of the UE doc I mentioned

**Comment by jstaerk** [comment]
These files are not Mustang's but the official rulesets from ZUGFeRD. If you send me an example which does not work but should and the list of missing codes I can upstream your request. 

Fun fact: they declared the "current" code lists valid but only publish them with every new ZUGFeRD release, so if your codes got added between the 2.2.1 release and now there is a certain likelyhood that they will be added anyway in the next version.

**Comment by cgimenez** [comment]
Oh I see... No unfortunately I don't have a working example yet - I've integrated Mustang in my Ruby test suite and stumble upon this anomaly, not a big deal but it would make sense for ZUGFeRD to integrate this VAT exception code (which is exactely the reason I need to use it : Repayment of expenditure), provided that it can be tracked on several documents amongst UE countries (eg. this one dated May 2023 : https://docs.peppol.eu/poacc/billing/3.0/codelist/vatex/)

So, if you're in touch with them would be worth to tell them that this specific code is missing

Thanks for your hints ;-)

**Comment by cgimenez** [comment]
About integrating Mustang it's very simple - my test suite exercice the whole invoicing stack, which at some point generates a Factur-x PDF. Then the test suite calls the Mustang executable as a cmd line process, capture the xml output, parse it then if errors are detected pretty print them and raise a test failure

About the missing VATEX-EU-79-C code that's not a big deal, even if it's a trick I use a different code in the test suite (one that don't make Mustang complaining) and VATEX-EU-79-C in production.

In the French Factur-x documentation reference (published by the FNFE and FeRD) the ExemptionReasonCode (whilst optionnal) is clearly mentionned as legitimate in the BASIC profile (which is the one I targeted)

About Peppol, I mentioned their page just to show that VATEX-EU-79-C is a legitimate code in the "nomenclature" of VATEX codes - not really Peppol related I must confess.

Nevermind thanks a lot to all people who maintain Mustang : the validator have been a life saver for me. It's a shame that the french side of Factur-x (the FNFE) don't promote it... In fact Mustang, AFAIK, is never mentioned anywhere in all the documents I have stubled upon french side.
